### PR TITLE
vehicle mounted pet carriers

### PR DESCRIPTION
**Pull request by mlangsdorf**
adds the infrastructure to support vehicle mounted pet carriers, and creates a new part, the livestock
carrier, that can be installed on a vehicle as a livestock stall and used as a vehicle mounted pet carrier.

### CODE REVIEW

**Comment by mlangsdorf** [comment]
Still to do:
1) Add weight of carried animal to vehicle/remove it when removed.
2) Remove cargo capacity when an animal is in the stall.
3) Add support for transporting large robots.

**Comment by tinukedaya** [comment]
I like this! But I would not say it fixes #11381 

It's very good and realistic approach for transport of large animals. But cats and dogs should be able to just be with you inside the car, without the need to put them to the stall (just like NPCs). For the small ones it still feels like a workaround, same as using Pet Carrier. Personally I just don't get any dogs anymore, even tho' I'd like to, they always ended up closed in the carrier, because it's just too much hassle to work with. 

BTW: I know this is a long stretch, but any chance this could be adapted to transport small vehicles? Bike rack, anyone? Or fork-lifter plugged at the back of my truck? 
Or maybe just one tile "garage" for the shopping cart, pallet lifter, or portable generator. I hate to always build the pallet lifter when I need to reinstall some storage batteries.

**Comment by mlangsdorf** [comment]
okay, it doesn't fix #11381. Fair enough. I am not currently up to untangling the NPC code that lets them ride in a vehicle enough to apply it to friendly monsters, though, so this is the work-around for now.

Repurposing this for vehicle racks: haha - wait, that might almost work.

So the monster capture routine works, in part, by grabbing the json description of the monster being captured and stashing it in the pet carrier, and then using that description to spawn a new monster with the exact same statistics when the pet carrier is emptied.

vehicles are described by json, there's no reason in principle that you couldn't do the same. Handling collisions would be challenging, but not impossible.

I want to do a rework of how actions are handled for vehicle parts in general, but it's possible that a vehicle rack could come out of that. Beyond the scope of this PR, though.

By the way, pallet lifters and telescopic cantilevers/telescoping cranes are foldable, you can build them on a folding frame with a unicycle wheel or castors and fold them up when not in use. The existing engine crane vehicle is exactly that design. So you don't need to do that. But it would be great to be able to transport ATVs on trucks and such.

**Comment by ZhilkinSerg** [comment]
Cannot capture friendly chicken

image

**Comment by mlangsdorf** [comment]
You're selecting the tile with the chicken in it? Because you're clearly within the iuse::capture_monster_act code, so you're failing the check:
```
if( const monster *const mon_ptr = g->critter_at<monster>( target ) ) {
```
which means you're aiming it at nothing.

I'll look at it more on Monday, but that's a weird failure mode because it's failing within the part of the code that I didn't change.

**Comment by mlangsdorf** [comment]
Okay, I summoned a chicken and fed it birdfood to make it friendly. 
chicken1
I examined the tile on my vehicle with the livestock stall
chicken2
then I selected the tile to the east. Didn't capture the chicken. Repeated the process, but selected the tile to the NE. Captured the chicken, and can see I have a chicken in the livestock stall.
chicken3a
chicken3
then I examined the tile again and released the critter.
chicken4

So I really can't reproduce your failure and I suspect you selected the wrong tile in your test.

**Comment by AMurkin** [comment]
Suggestion: add option to transfer pet from/to regular pet carrier.

**Comment by ZhilkinSerg** [comment]
> So I really can't reproduce your failure and I suspect you selected the wrong tile in your test.

I guess I know what was wrong in my test - I've pressed key to select tile relative to vehicle part position instead of player position. Will try to test again tonight.

**Comment by mlangsdorf** [comment]
Yes, it's position relative to the player. Thanks for the retest.

File: src/vehicle.cpp [hunk 1]
```diff
@@ -6760,3 +6769,22 @@ void vehicle::use_washing_machine( int p ) {
...
+{
+    if( parts[part].is_broken() || parts[part].removed ) {
+        return;
+    }
+    item base = item( parts[part].get_base() );
+    base.type->invoke( g->u, base, pos );
+    parts[part].set_base( base );
+    /* captured animals take up all the cargo space */
+    /*
+    if( base.has_var( "contained_name" ) ) {
```
Comment (Night-Pryanik) [inline src/vehicle.cpp]: I guess this commented text should be deleted.

File: doc/JSON_FLAGS.md [hunk 2]
```diff
@@ -467,6 +467,7 @@ These branches are also the valid entries for the categories of `dreams` in `dre
 - ```BOARDABLE``` The player can safely move over or stand on this part while the vehicle is moving.
 - ```CAMERA```
 - ```CAMERA_CONTROL```
+- ```CAPTURE_MOSNTER_VEH``` Can be used to capture monsters when mounted on a vehicle.
```
Comment (Night-Pryanik) [inline doc/JSON_FLAGS.md]: Typo.

Comment (mlangsdorf) [reply doc/JSON_FLAGS.md]: What's the typo? I'm not seeing it.

Comment (Night-Pryanik) [reply doc/JSON_FLAGS.md]: `MOSNTER`

**Pull request merged**